node/src/actors/chain_manager/mod.rs

```diff
@@ -1034,10 +1041,10 @@ impl ChainManager {
                             "Consensus while sync! Superblock {:?}",
                             act.get_superblock_beacon()
                         );
-                        // Persist current_chain_state with
-                        // current_superblock_state and the new superblock beacon
-                        act.last_chain_state = act.chain_state.clone();
-                        act.persist_chain_state(ctx);
+
+                        // Copy current chain state into previous chain state, and persist it
+                        act.advance_chain_state();
+                        act.persist_previous_chain_state(ctx);
```

Comment:
Here you are moving current_chain_state to previous and persist previous, so... you are persisting the current chain_state. I think it should be first persist the previous one and update the new one as previous
```suggestion
                        act.persist_previous_chain_state(ctx);
                        act.advance_chain_state();
```